modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc

```diff
@@ -0,0 +1,403 @@
+
+The assisted service pod converts `registries.conf` into an `imageContentSourcesPolicy` file using the `--icsp` flag in some `oc adm` commands. These `oc adm` commands are run against the release image, which is executed from within the assisted service pod.
+
+The `ca-bundle.crt` and `registries.conf` keys can be added individually or together.
+
+[NOTE]
+====
+The ConfigMap should be installed in the same namespace as the infrastructure-operator (ie. `assisted-installer`).
+====
+
+  . To configure the mirror registry, first create and upload the ConfigMap containing the `ca-bundle.crt` and `registries.conf` keys:
+
+[source,terminal]
+----
+$ cat <<EOF | kubectl create -f -
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: mirror-registry-config-map
+  namespace: "assisted-installer"
+  labels:
+    app: assisted-service
+data:
+  ca-bundle.crt: |
+    -----BEGIN CERTIFICATE-----
+    certificate contents
+    -----END CERTIFICATE-----
+  registries.conf: |
+    unqualified-search-registries = ["registry.access.redhat.com", "docker.io"]
+    [[registry]]
+       prefix = ""
+       location = "quay.io/edge-infrastructure"
+mirror-by-digest-only = true
+       [[registry.mirror]]
+       location = "mirror1.registry.corp.com:5000/edge-infrastructure"
+EOF
+----
+
+Registries listed in the `unqualified-search-registries` will be automatically added to an authentication ignore list (`PUBLIC_CONTAINER_REGISTRIES` environment variable) and will not be required by `assisted-service` when validating the pull secret.
+Registries defined in the `registries.conf` must be scoped by repository and not by registry.
+
+[NOTE]
+====
+`quay.io/edge-infrastructure` and `mirror1.registry.corp.com:5000/edge-infrastructure` are both scoped by the `edge-infrastructure` repository. This is a valid configuration.
+
+Removing the repository `edge-infrastructure` from location is an invalid configuration, as it will not pass openshift-installer validation.
+====
+
+  . Set the `mirrorRegistryRef` in the spec of `AgentServiceConfig` to the name of uploaded `ConfigMap` as follows:
+
+[source,terminal]
+----
+$ cat <<EOF | kubectl apply -f -
+apiVersion: agent-install.openshift.io/v1beta1
+kind: AgentServiceConfig
+metadata:
+  name: agent
+spec:
+  databaseStorage:
+    accessModes:
+    - ReadWriteOnce
+    resources:
+      requests:
+        storage: 10Gi
+  filesystemStorage:
+    accessModes:
+    - ReadWriteOnce
+    resources:
+      requests:
+        storage: 20Gi
+  imageStorage:
+    accessModes:
+    - ReadWriteOnce
+    resources:
+      requests:
+        storage: 10Gi
+  mirrorRegistryRef:
+    name: mirror-registry-config-map
+EOF
+----
+
+For more details on how to specify the CR, see [AgentServiceConfig CRD](https://github.com/openshift/assisted-service/blob/master/config/crd/bases/agent-install.openshift.io_agentserviceconfigs.yaml).
```

Comment:
To generate a link here, you have to use the following syntax:
`link:<https://example.com>[Example Title]`

(Btw, I think I see the same problem on [line 179](https://github.com/openshift/openshift-docs/pull/55352/files#diff-409f6af30b90a6d04d6ba57f8de6ffe514553beb7f7c14717e92bd09dcc614e5R179))

Reply:
Updated as suggested.